t/25-cache.t

```diff
@@ -0,0 +1,247 @@
+                  if $filename =~ /sle-12-SP3-x86_64-0368-404/;
+                return $c->render(status => 400, text => "Move along, nothing to see here")
+                  if $filename =~ /sle-12-SP3-x86_64-0368-400/;
+                return $c->render(status => 500, text => "Move along, nothing to see here")
+                  if $filename =~ /sle-12-SP3-x86_64-0368-589/;
+                if ($filename =~ /sle-12-SP3-x86_64-0368-589/) {
+                    $c->res->headers->content_length(10);
+                    $c->write(
+                        'Hel' => sub {
+                            my $c = shift;
+                            $c->write('lo!');
+                        });
+                }
+            });
+
+        $mock->routes->get(
+            '/' => sub {
+                my $c = shift;
+                return $c->render(status => 200, text => "server is running");
+            });
+
+        # Connect application with web server and start accepting connections
+
+        $daemon = Mojo::Server::Daemon->new(app => $mock, listen => [$host]);
+        $daemon->start;
+    }
+}
+
+start_server;
+Mojo::IOLoop->start if !Mojo::IOLoop->is_running;
+
+chdir $ENV{LOGDIR};
+get_asset({id => 922756}, "hdd", 'sle-12-SP3-x86_64-0368-textmode@64bit.qcow2');
+
+my $autoinst_log = read_log('autoinst-log.txt');
+
+like $autoinst_log, qr/Downloading sle-12-SP3-x86_64-0368-textmode\@64bit.qcow2 from/, "Asset download attempt";
+like $autoinst_log, qr/failed with: 521/, "Asset download fails with connection refused";
+truncate_log;
+
+get_asset({id => 922756}, "hdd", 'sle-12-SP3-x86_64-0368-404@64bit.qcow2');
+$autoinst_log = read_log('autoinst-log.txt');
+like $autoinst_log, qr/Downloading sle-12-SP3-x86_64-0368-404\@64bit.qcow2 from/, "Asset download attempt";
+like $autoinst_log, qr/failed with: 404/, "Asset download fails with connection refused 404";
+truncate_log;
+
+get_asset({id => 922756}, "hdd", 'sle-12-SP3-x86_64-0368-400@64bit.qcow2');
+$autoinst_log = read_log('autoinst-log.txt');
+like $autoinst_log, qr/Downloading sle-12-SP3-x86_64-0368-400\@64bit.qcow2 from/, "Asset download attempt";
+like $autoinst_log, qr/failed with: 400/, "Asset download fails with connection refused 400";
+truncate_log;
+
+get_asset({id => 922756}, "hdd", 'sle-12-SP3-x86_64-0368-500@64bit.qcow2');
+$autoinst_log = read_log('autoinst-log.txt');
+like $autoinst_log, qr/Downloading sle-12-SP3-x86_64-0368-500\@64bit.qcow2 from/, "Asset download attempt";
+like $autoinst_log, qr/failed with: 500/, "Asset download fails with connection refused 500";
+truncate_log;
+
+get_asset({id => 922756}, "hdd", 'sle-12-SP3-x86_64-0368-589@64bit.qcow2');
+$autoinst_log = read_log('autoinst-log.txt');
+like $autoinst_log, qr/Downloading sle-12-SP3-x86_64-0368-589\@64bit.qcow2 from/, "Asset download attempt";
+like $autoinst_log, qr/failed with: 589/, "Asset download fails with connection refused 589";
+truncate_log;
+
+Mojo::IOLoop->stop;
```

Comment:
This line does nothing either and maybe should be replaced with code to stop the forked test server. Something you can find in the [Mojolicious tests](https://github.com/kraih/mojo/blob/master/t/mojo/morbo.t#L140-L142) too.
```perl
# Stop
kill 'INT', $test_server_pid;
sleep 1 while _port($port);
```